src/Radicle/Internal/Json.hs

```diff
@@ -0,0 +1,91 @@
+module Radicle.Internal.Json where
+
+import           Protolude
+
+import           Control.Monad.Fail
+import qualified Data.Aeson as A
+import qualified Data.Aeson.Types as A
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Map.Strict as Map
+import qualified Data.Sequence as Seq
+import qualified Data.Vector as JsonVector
+
+import           Radicle.Internal.Core
+import           Radicle.Internal.Identifier
+import           Radicle.Internal.Number
+import           Radicle.Internal.Parse
+import           Radicle.Internal.Pretty
+import qualified Radicle.Internal.Type as Type
+
+-- | Parses JSON into a Value. It expects the value to be encoded as a radicle
+-- expression in a JSON string. Corresponds to the encoding given by
+-- 'valueToJson'.
+jsonToValue :: A.Value -> A.Parser Value
+jsonToValue = A.withText "Value" $ \t -> do
+    case parse "[daemon]" t of
+      Left err -> fail $ "failed to parse Radicle expression: " <> show err
+      Right v  -> pure v
+
+-- | Encodes a radicle value as JSON. The value is encoded a radicle expression
```

Comment:
Don't know how to do multiline suggestions, but maybe instead:
```
-- | Pretty-prints a Radicle value, and wraps it in a JSON string
```
So it's clear this isn't in "deeper" way encoding Rad as JSON.

Reply:
And maybe also a doctest example